tests/php/sync/test_class.jetpack-sync-posts.php

```diff
@@ -45,41 +43,53 @@ public function test_add_post_syncs_post_data() {
 
 	public function test_add_post_syncs_request_is_auto_save() {
 		//Sync from setup should not be auto save
-		$event = $this->server_event_storage->get_most_recent_event( 'wp_insert_post' );
+		$event = $this->server_event_storage->get_most_recent_event( 'jetpack_save_post' );
 		$this->assertFalse( $event->args[3] );
-		
+
 		Jetpack_Constants::set_constant( 'DOING_AUTOSAVE', true );//define( 'DOING_AUTOSAVE', true );
 
 		//Performing sync here (even though setup() does it) to sync REQUEST_URI
 		$user_id = $this->factory->user->create();
 		$this->factory->post->create( array( 'post_author' => $user_id ) );
 		$this->sender->do_sync();
 
-		$event = $this->server_event_storage->get_most_recent_event( 'wp_insert_post' );
+		$event = $this->server_event_storage->get_most_recent_event( 'jetpack_save_post' );
 		$this->assertTrue( $event->args[3] );
 	}
 
 	public function test_trash_post_trashes_data() {
 		$this->assertEquals( 1, $this->server_replica_storage->post_count( 'publish' ) );
-
+		$this->server_event_storage->reset();
 		wp_delete_post( $this->post->ID );
 
 		$this->sender->do_sync();
-		$event = $this->server_event_storage->get_most_recent_event( 'jetpack_trashed_post' );
+		$insert_event = $this->server_event_storage->get_most_recent_event( 'jetpack_save_post' );
+
+		$this->assertEquals( $insert_event->args[1]->post_status, 'trash' );
+		$this->assertEquals( $insert_event->args[0], $this->post->ID );
 
-		$this->assertEquals( 'post', $event->args[1] );
-		$this->assertTrue( (bool) $event );
 		$this->server_event_storage->reset();
 
 		$this->assertEquals( 0, $this->server_replica_storage->post_count( 'publish' ) );
 		$this->assertEquals( 1, $this->server_replica_storage->post_count( 'trash' ) );
-
 		wp_delete_post( $this->post->ID );
 		$this->sender->do_sync();
-		
+
 		// Since the post status is not changing here we don't expect the post to be trashed again.
-		$event = $this->server_event_storage->get_most_recent_event( 'jetpack_trashed_post' );
-		$this->assertFalse( (bool) $event );
+		$delete_event = $this->server_event_storage->get_most_recent_event( 'deleted_post' );
+		$save_event = $this->server_event_storage->get_most_recent_event( 'jetpack_save_post' );
+		$this->assertFalse( $save_event );
+		$this->assertTrue( (bool) $delete_event );
+	}
+
+	public function test_sync_post_event_includes_previous_state() {
+		// $this->assertEquals( 1, $this->server_replica_storage->post_count( 'publish' ) );
+		$this->server_event_storage->reset();
+		wp_delete_post( $this->post->ID );
+		$this->sender->do_sync();
+		$insert_event = $this->server_event_storage->get_most_recent_event( 'jetpack_save_post' );
+		$this->assertEquals( 'trash', $insert_event->args[1]->post_status ); //
```

Comment:
Is one of these lines (either 91 or 92) the current state and one of them the previous state? If so, can we identify them in the object as such?